src/Common/Process.cpp

```diff
@@ -75,58 +100,93 @@ static ProcessID StartProcess( const std::wstring_view CommandLine, FILE* pOutpu
 		dup2( fileno( pOutputStream ), 1 );
 		dup2( fileno( pOutputStream ), 2 );
 
-		execl( "/bin/sh", "/bin/sh", "-c", UTF8Converter().to_bytes( CommandLine.data(), CommandLine.data() + CommandLine.size() ).c_str(), NULL );
+		execl( "/bin/sh", "/bin/sh", "-c", UTF8Converter().to_bytes( m_CommandLine.data(), m_CommandLine.data() + m_CommandLine.size() ).c_str(), NULL );
 
 		exit( EXIT_FAILURE );
 	}
 
-	return PID;
+	m_Pid = PID;
 
 #endif // __linux__ || __APPLE__
 
-} // StartProcess
+} // Start
 
 //////////////////////////////////////////////////////////////////////////
 
-static int WaitProcess( ProcessID PID )
+int Process::Wait( void )
 {
 
 #if defined( _WIN32 )
 
 	BOOL  Result;
 	DWORD ExitCode;
 
-	while( WIN32_CALL( Result = GetExitCodeProcess( PID, &ExitCode ) ) && ExitCode == STILL_ACTIVE )
+	while( WIN32_CALL( Result = GetExitCodeProcess( m_Pid, &ExitCode ) ) && ExitCode == STILL_ACTIVE )
 		Sleep( 1 );
 
-	CloseHandle( PID );
+	CloseHandle( m_Pid );
 
-	return Result ? static_cast< int >( ExitCode ) : -1;
+	m_Pid = nullptr;
+	m_ExitCode = ExitCode;
+
+	return Result ? static_cast< int >( m_ExitCode ) : -1;
 
 #elif defined( __linux__ ) || defined( __APPLE__ ) // _WIN32
 
 	int status;
-	waitpid( PID, &status, 0 );
+	waitpid( m_Pid, &status, 0 );
+
+	m_ExitCode = status;
+
+	Kill();
+
+	return m_ExitCode;
+
+#endif // __linux__ || __APPLE__
+
+} // Wait
+
+//////////////////////////////////////////////////////////////////////////
+
+void Process::Kill( void )
+{
+
+#if defined( _WIN32 )
+
+	//CloseHandle( m_Pid );
+	TerminateProcess( m_Pid, 1 );
+
+	if( m_ExitCode < 0 )
+		m_ExitCode = 1;
+
+	m_Pid = nullptr;
+
+#elif defined( __linux__ ) || defined( __APPLE__ ) // _WIN32
+
+	execl( "/bin/sh", "/bin/sh", "kill", "-9", m_Pid );
```

Comment:
Are we going to run into permission issues by trying to kill a process from the command line?